### DISCUSSION TITLE
Error on IBM i Debug

### DISCUSSION THREAD

**Discussion by viruslogic** [codefori/vscode-ibmi#1202] [Q&A] [answered]
Hi,

My name is Marcos and I'm an AS400 developer from Spain. I've been using all the necessary extensions to develop from VS Code for a long time and I'm excited about it.

Recently I read the news that an extension was available to debug from VS Code.  I've been following all the installation steps: PTFs, Certificates, Services, etc. 

The thing is that when I go to start the debug I get an error and I don't know what to do.

The message I get is: "You are not authorised to program IRPG301 in library IDEA. 

![image](https://user-images.githubusercontent.com/10233590/228839665-a64ca895-dcd0-4723-9916-addefeb1a441.png)

The problem is that the source is in one library and the object is in another. If I try to run debug to start from the library where the object is, it still doesn't work. 

But if I copy the object to the library where the source is, it works.

Can someone give me a hand? 

Thanks!

**Comment by worksofliam** [comment, marked as answer]
Closing as spam. https://github.com/halcyon-tech/vscode-ibmi/issues/1203
